module.h

```diff
@@ -479,6 +480,55 @@ namespace ws::module {
 
 
 
+    struct Token {
+        std::string content;
+        std::uint32_t type;
+        std::uint64_t line;
+        std::uint64_t column;
+    };
+
+    /*
+        8 bytes: content's size
+        <size> bytes: content
+        4 bytes: type
+        8 bytes: line
+        8 bytes: column
+    */
+    inline std::ostream& serialize(std::ostream& os, Token const& token) {
+        std::size_t size = token.content.size();
+        os.write(reinterpret_cast<const char*>(&size               ), sizeof(size        ));
+        os.write(                               token.content.data(),        size         );
+        os.write(reinterpret_cast<const char*>(&token.type         ), sizeof(token.type  ));
+        os.write(reinterpret_cast<const char*>(&token.line         ), sizeof(token.line  ));
+        os.write(reinterpret_cast<const char*>(&token.column       ), sizeof(token.column));
+        return os;
```

Comment:
Would it be possible to just reinterpret cast the _entire_ struct into chars and output that?
```suggestion
        os.write(reinterpret_cast<const char*>(&token), sizeof(token))
```

Reply:
no cause of the content

Reply:
If you reorder the members so that the primitives come first, you could probably cast them all in one call then do the string separate.